pika/connection.py

```diff
@@ -1593,10 +1642,8 @@ def _on_terminate(self, reason_code, reason_text):
         for channel in dictkeys(self._channels):
             if channel not in self._channels:
                 continue
-            method_frame = frame.Method(channel, spec.Channel.Close(
-                reason_code,
-                reason_text))
-            self._channels[channel]._on_close(method_frame)
+            # pylint: disable=W0212
+            self._channels[channel]._on_close_meta(reason_code, reason_text)
```

Comment:
REVIEWME now calls the dedicated channel method `_on_close_meta` instead of overloading the logic of `_on_close`, which should be dedicated to handling `Channel.Close` frames from broker.